src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -430,16 +431,17 @@ impl Coordinator {
         timeline_context: TimelineContext,
         real_time_recency_ts: Option<mz_repr::Timestamp>,
     ) -> Result<TimestampDetermination<mz_repr::Timestamp>, AdapterError> {
+        let isolation_level = session.vars().transaction_isolation();
         let det = self.determine_timestamp_for(
             self.catalog().state(),
             session,
             id_bundle,
             when,
             compute_instance,
-            timeline_context,
+            timeline_context.clone(),
```

Comment:
This is only used as a ref at the lowest level, so should change to that throughout.